accounts-db/src/accounts_db.rs

```diff
@@ -5404,6 +5404,68 @@ impl AccountsDb {
         )
     }
 
+    /// Load account with `pubkey`. `callback` is used to decide whether to
+    /// store the account into read cache after loading.
+    ///
+    /// Return the account and the slot when the account was last stored.
+    /// Return None for ZeroLamport accounts.
+    pub fn load_account_with(
```

Comment:
The pattern for these two functions is similar. But the function signatures are very different. Because of the function signature difference, almost every code if branches inside the function body is a little different. this makes sharing the code difficult. Therefore, I think it is probaby okay to keep them separate. WDYT?

Reply:
We could potentially make `do_load_with_populate_read_cache(key)` a alias for `load_account_with(key, |_| true)`.